app/code/Magento/Customer/view/frontend/web/js/show-password.js

```diff
@@ -0,0 +1,59 @@
+/**
+* Copyright © Magento, Inc. All rights reserved.
+* See COPYING.txt for license details.
+*/
+
+define([
+    'jquery',
+    'ko',
+    'uiComponent'
+], function ($, ko, Component) {
+    'use strict';
+
+    return Component.extend({
+        passwordSelector: '',
+        showPasswordSelector: '[data-role=show-password]',
+        passwordInputType: 'password',
+        textInputType: 'text',
+
+        defaults: {
+            template: 'Magento_Customer/show-password'
+        },
+
+        /** @inheritdoc */
+        initialize: function () {
+            this._super();
+        },
+
+        /**
+         * @return {Object}
+         */
+        initObservable: function () {
+            var self = this;
+
+            this._super()
+                .observe({
+                    isChecked: ko.observable(false)
+                });
```

Comment:
I'd declare `isChecked` in the options defaulting to `false` and use the following syntax here:
```suggestion
            this._super()
                .observe([
                    'isChecked'
                ]);
```
That should allow removing `ko` from the parameters. Also, it may be better to abstract from specific element types in UI component (checkbox) and rename `isChecked` option to `isPasswordVisible`.

Reply:
I have removed the ko parameter and also renamed it to `isPasswordVisible`. Updated the code based on your suggestions.